metricflow/dataflow/nodes/compute_metrics.py

```diff
@@ -28,9 +28,11 @@ class ComputeMetricsNode(DataflowPlanNode):
     metric_specs: Tuple[MetricSpec, ...]
     for_group_by_source_node: bool
     _aggregated_to_elements: Tuple[LinkableInstanceSpec, ...]
+    _alias_to_metric_spec: Dict[str, MetricSpec] = field(hash=False)
```

Comment:
I added this private thing as a way to cache the `alias -> MetricSpec` map, mostly because if we compare multiple times against the same node we don't need to iterate over all of its metric specs.

Not really sure if this optimization is necessary, though. We might be able to get away with just looping through things everytime.

Reply:
In this case, there can't be multiple metrics associated with each alias, though. In one compute metrics node, there can only be one alias per metric since we're assuming metrics need to be unique within one derived metric. So keeping it as a dict is fine, I think.

Do you think we need this dict or should I just go back to a for loop and avoid this caching?

Reply:
Ohhh good point! I don't think we should keep it as a param that can be overridden. I'm not worried about the list of input metrics being long so yeah I'm fine with just putting this logic where it will be used.

Reply:
I removed this logic from the dataclass, and now we're building the dict inside `can_combine`.